assets/src/components/highlightedMatch.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react"
+import { intersperseString } from "../helpers/array"
+import { filterToAlphanumeric } from "../models/searchQuery"
+
+export const HighlightedMatch = ({
+  content,
+  highlightText,
+}: {
+  content: string
+  highlightText?: string
+}): JSX.Element => {
+  if (highlightText === undefined) {
+    return <>{content}</>
+  }
+
+  const match = content.match(highlightRegex(highlightText))
+
+  if (match === null || match.index === undefined) {
+    return <>{content}</>
+  }
+
+  const matchingString = match[0]
+
+  return (
+    <>
+      {[
+        content.slice(0, match.index),
+        <span className="highlighted" key={`highlighted-${match.index}`}>
+          {matchingString}
+        </span>,
+        <HighlightedMatch
+          content={content.slice(match.index + match[0].length)}
+          highlightText={highlightText}
+          key={`highlighted-extension-${match.index + match[0].length}`}
+        />,
+      ]}
+    </>
+  )
+}
```

Comment:
__question__(non-blocking): Could this be done without recursion? I was finding it a bit hard to follow at first, and I'm partially worried about JS/react having difficulty with this pattern, though less so than the readability.

Reply:
It probably is doable, though for the purposes of this PR I just wanted to move the old code to a more sensible place.

Reply:
Oh oops, I hadn't noticed that this was moving code and not new code.